modules/api/test/LorisApiDicomsTest.php

```diff
@@ -0,0 +1,362 @@
+            'SeriesUID',
+            $DicomArray['DicomTars']['0']['SeriesInfo']['0']
+        );
+    }
+
+    /**
+     * Tests the HTTP POST request for the
+     * endpoint /candidates/{candid}/{visit}/dicoms
+     * // TODO HANDLING OF POST HANDLING NOT IMPLEMENTED
+     *
+     * @return void
+     */
+    public function testPostCandidatesCandidVisitDicoms(): void
+    {
+        $response = $this->client->request(
+            'POST',
+            "candidates/$this->candidTest/$this->visitTest/dicoms",
+            [
+                'headers'     => $this->headers,
+                'http_errors' => false,
+                'json'        => []
+            ]
+        );
+        if ($response->getStatusCode() === 405) {
+            $this->markTestIncomplete(
+                "The method POST is not available for the" .
+                "endpoint candidates/$this->candidTest/$this->visitTest/dicoms"
+            );
+        }
+        // Verify the status code - TODO METHOD NOT ALLOWED YET
+        $this->assertEquals(200, $response->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response->getBody();
+        $this->assertNotEmpty($body);
+
+    }
+
+    /**
+     * Tests the HTTP GET request for the
+     * endpoint /candidates/{candid}/{visit}/dicoms/tarname
+     *
+     * @return void
+     */
+    public function testGetCandidatesCandidVisitDicomsTarname(): void
+    {
+        $response = $this->client->request(
+            'GET',
+            "candidates/$this->candidTest/$this->visitTest/" .
+            "dicoms/$this->tarfileTest",
+            [
+                'headers'     => $this->headers,
+                'http_errors' => false
+            ]
+        );
+        if ($response->getStatusCode() === 404) {
+            $this->markTestIncomplete(
+                "Endpoint not found: " .
+                "candidates/$this->candidTest/$this->visitTest/dicoms "
+            );
+        }
+        $this->assertEquals(200, $response->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response->getBody();
+        $this->assertNotEmpty($body);
+
+        $resource        = fopen($this->tarfileTest, 'w');
+        $stream          = GuzzleHttp\Psr7\stream_for($resource);
+        $response_stream = $this->client->request(
+            'GET',
+            "candidates/$this->candidTest/$this->visitTest/dicoms/" .
+            "$this->tarfileTest",
+            [
+                'http_errors' => false,
+                'headers'     => $this->headers,
+                'save_to'     => $stream,
+            ]
+        );
+        $this->assertEquals(200, $response_stream->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response_stream->getBody();
+        $this->assertNotEmpty($body);
+
+        $this->assertFileIsReadable($this->tarfileTest);
+    }
+
+    // THESE ENDPOINTS DO NOT EXIST YET, this is why the tests assert
```

Comment:
If the endpoints do not exists, remove the tests.

Reply:
ok